src/player.cpp

```diff
@@ -6616,10 +6616,13 @@ void player::process_active_items()
         } else if( identifier == "adv_UPS_off" ) {
             ch_UPS += it.ammo_remaining() / 0.6;
         }
-        if( !it.has_flag( "USE_UPS" ) && it.charges < it.type->maximum_charges() ) {
+        if( it.has_flag( "USE_UPS" ) && it.charges < it.type->maximum_charges() ) {
             active_held_items.push_back( index );
         }
     }
+    if( has_active_bionic( bionic_id( "bio_ups" ) ) ) {
+        ch_UPS += power_level;
+    }
```

Comment:
Shouldn't think be in the loop above, so normal items see the bionic UPS as a valid supply to draw from? If I understand the bionic UPS description correctly, this should work fine.

Reply:
This is just for getting the sum of all the UPS charges.